### ISSUE TITLE
Compiler crash related to `Tuple in eval_single`

### ISSUE THREAD

**Issue by PROMETHIA-27** [jfecher/ante#153]
Unsure exactly what's happening here but the following code produces a compiler crash:
```
type Box t = ptr: Ptr t

box (value: t) = 
    ptr = malloc (size_of (MkType : Type t))
    ptr_store ptr value
    Box ptr

unbox (value: Box t) =
    deref_ptr value.ptr

type BoxedAny = 
    id: Usz
    data: Box Unit

into_any_box (value: t) = 
    BoxedAny 
        (type_id (MkType : Type t)) 
        (Box (transmute (box value).ptr))

downcast (box: BoxedAny) : Maybe (Box t) = 
    if type_id (MkType : Type t) == box.id 
        then Some (transmute box.data)
        else None

value = 12

boxed_value: BoxedAny = into_any_box value

str: Box String = boxed_value
    |> downcast
    |> unwrap
    |> unbox
```
it appears to be a mostly minimal repro, I was unable to remove or simplify without losing the crash.
Error:
`thread 'main' panicked at 'Value::Tuple found in eval_single: [Normal(v0)]', src\cranelift_backend\context.rs:115:36`

**Comment by jfecher** [comment]
Looks like this is a result of structs/tuples not being supported for the builtin transmute function when using the cranelift backend. I think it is fixable if when a struct value is found we fall back to a ReinterpretCast instead.

**Comment by jfecher** [comment]
Fixed in 700b1b5
